### PR TITLE
1383 fix on client side

### PR DESCRIPTION
**Pull request by elcazador**
Revised fix for issue 1383. Bot/crawler detection now occurs on client
side using most common known bots and crawlers.  This will not exclude
other crawlers.  Using a blacklist is probably desirable to whitelisting browsers and having to spend time maintaining the white list when complaints come in.

Made minor change to have the survey panel be hidden by default and only
shown when needed.  This fixed a screen flicker that happened when the
popup was rendered and then hidden on slower machines.

### CODE REVIEW

File: js/popup_survey.js [hunk 1]
```diff
@@ -6,12 +6,18 @@
 
   $(document).ready(function() {
     centrePopup();
+    var bots = /alexa|bot|crawl|facebookexternalhit|feedburner|google web preview|nagios|postrank|pingdom|slurp|spider|yahoo!|yandex|sogou/i;
+
+    if( navigator.userAgent.match(bots) ) {
```
Comment (minorOffense) [inline js/popup_survey.js]: Only thing I would add is to check that navigator.userAgent isn't undefined. I remember something about that being deprecated or changing or something...

File: popup_survey.module [hunk 2]
```diff
@@ -174,12 +176,15 @@ function popup_survey_bean_options() {
     ->entityCondition('entity_type', 'bean', '=')
     ->propertyCondition('type', 'survey_popup_message', '=');
   $result = $query->execute();
-  
+
   // From the query result pull out each of the bean ids.
   $ids = array();
-  foreach($result['bean'] as $record) {
-    $ids[] = $record->bid;
+  if( count($result) > 0 ) {
```
Comment (minorOffense) [inline popup_survey.module:203]: I'd do an isset or is_array on bean instead

**Review by minorOffense — CHANGES_REQUESTED** [review]

File: popup_survey.module (hunk 2 above)
Comment (elcazador) [reply popup_survey.module:203]: I tried isset first and then is_array but the query result is always an array, just an empty one so they  always returns true.  This is a check for the existence of data, not a failed query.

**Review by minorOffense — APPROVED** [review]

File: popup_survey.module [hunk 3]
```diff
@@ -54,88 +54,111 @@ function popup_survey_admin_settings() {
...
+      $ua = check_plain(variable_get('popup_survey_ua_exclude',''));
+      drupal_add_js(array('POPUP_SURVEY' => array('botlist' => $ua)), array('type' => 'setting'));
+
       $attached['css'][$path . '/css/popup_survey.css'] = $options;
       $attached['js'][$path . '/js/popup_survey.js'] = $options;
       $bean = entity_view('bean', entity_load('bean', array($bid)));
       $page['page_bottom']['popup_survey']['#markup'] = '<div class="popup-greyout">' . render($bean) . '</div>';
-  
+
       // DEBUG: dpm($page['page_bottom']['popup_survey']);
```
Comment (minorOffense) [inline popup_survey.module:159]: should remove this dpm even if it's commented out.

**Pull request closed without merging**